src/lib/common/win32/KnownFolderPaths.h

```diff
@@ -0,0 +1,48 @@
+/*
+* barrier -- mouse and keyboard sharing utility
+* Copyright (C) 2018 Debauchee Open Source Group
+*
+* This package is free software; you can redistribute it and/or
+* modify it under the terms of the GNU General Public License
+* found in the file LICENSE that should have accompanied this file.
+*
+* This package is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+#include "StringConverters.h"
+#include <string>
+#include <Shlobj.h>
+
+static std::string known_folder_path(const KNOWNFOLDERID& id)
```

Comment:
`static` is incorrect for functions that are in the headers and are supposed to be reused in many cpp files. In such case there's a separate copy of the function in each translation unit and we rely on the linker to deduplicate.

Better to use `inline`.

Even better would be to add a .cpp file for the implementations.

The same applies to a header below.